CONTRIBUTING.md

```diff
@@ -35,22 +35,54 @@ Contributions are essential for keeping this extension great. We try to keep it
 	```bash
 	$ mvnw.cmd verify
 	```
-       
+
 
 5. `cd vscode-xml/`
-   
-6. Install the dependencies:  
-	```bash  
+
+6. Install the dependencies:
+	```bash
 	$ npm install
 	```
 
 7. In `vscode-xml/`, build the server by running:
 
-	```bash   
+	```bash
 	$ npm run build-server
 	```
 
 8. To run the extension, open the Debugging tab in VSCode.
 9. Select and run 'Launch Extension (vscode-xml)' at the top left:
 
     ![ Launch Extension ](./images/LaunchExtension.png)
+
+## Binary server development
+
+### Testing a binary version of LemMinX
+
+1. Copy the binary version of LemMinX to:
+
+   | OS | Location (relative to root of repository) |
+   | --- | --- |
+   | Linux | `./server/lemminx-linux` |
+   | macOS | `./server/lemminx-darwin-x86_64` |
+   | Windows | `.\server\lemminx-win32.exe` |
+
+2. Make sure that you set `xml.server.preferBinary` to `true`,
+and disable any [LemMinX extensions](./docs/Extensions.md)
+by commenting out `xml.extension.jars` in your `settings.json`
+and uninstalling or disabling any VS Code extensions that provide extra LemMinX features.
+
+3. Launch vscode-xml in development mode, and double check that the binary server is running by checking the server logging (Output > XML Support)
+
+### Building a binary with GraalVM locally
+
+If you submit a LemMinX PR, GitHub Actions will generate a binary for your PR that you can use for testing.
+If you need to generate a LemMinX binary locally for whatever reason, follow these steps:
+
+1. Download and set up [GraalVM and its dependencies](https://www.graalvm.org/docs/getting-started/)
+    * either 20.x or 21.x should work
+    * The version for Java 11 is preferred, since it is hard to set up GraalVM Java 8 on Windows
+2. Make sure the executables `java` and `javac`, and the environment variable `JAVA_HOME` all point to the GraalVM installation.
```

Comment:
I'm pretty sure this step just requires `JAVA_HOME` to point to the GraalVM installation.